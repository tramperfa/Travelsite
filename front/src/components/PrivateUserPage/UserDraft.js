import React from 'react';
import Button from 'material-ui/Button';
import PropTypes from 'prop-types';
import {Redirect} from 'react-router-dom';
import {compose} from 'recompose';

//
import {WithCreateDraftMutation, WithDraftListQuery} from '../../graphql/draft';
import ComposeQuery from '../../lib/hoc';
import ErrorComp from '../../lib/ErrorComp';
import {error, onError} from '../../lib/utils';

//
import DraftCard from './DraftCard';

export class UserDraft extends React.Component {
	state = {
		newDraftID: null,
		error: error
	}

	handleCreate = () => {
		this.props.createDraft().then((data) => {
			this.setState({newDraftID: data.data.createDraft._id})
		}).catch((err) => {
			this.setState(onError(err))
		})
	}

	render() {
		if (this.state.newDraftID) {
			return <Redirect push={true} to={`/my/edit/${this.state.newDraftID}`}/>;
		}

		console.log(this.props.draftList.myDrafts);

		return (
			<div>
				<ErrorComp error={this.state.error}/>
				<div>
					{
						this.props.draftList.myDrafts.map(draft => (
							<div key={draft._id} className='draft'>
								<DraftCard draft={draft}/>
							</div>
						))
					}
				</div>
				<div className='create'>
					<Button
						raised={true}
						color="primary"
						onClick={this.handleCreate}
						label='draftCreate'>
						CREATE NEW DRAFT
					</Button>
				</div>

			</div>
		)
	}
}

UserDraft.propTypes = {
	createDraft: PropTypes.func.isRequired,
	draftList: PropTypes.object.isRequired
}

const UserDraftWithComposeQuery = ComposeQuery(UserDraft, 'draftList')

export default compose(WithDraftListQuery, WithCreateDraftMutation)(
	UserDraftWithComposeQuery
)
