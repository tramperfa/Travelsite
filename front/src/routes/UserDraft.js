import React from 'react'
import {graphql} from 'react-apollo';
import Button from 'material-ui/Button';
import PropTypes from 'prop-types';
import {Redirect} from 'react-router-dom';

//
import {draftsListQuery, createDraftMutation} from '../graphql/draft';

//
import DraftList from '../components/DraftList';

export class TheUserDraft extends React.Component {
	state = {
		newDraftID: null
	}

	handleCreate = async () => {
		this.props.createDraft().then((data) => {
			//console.log(JSON.stringify(data));
			this.setState({newDraftID: data.data.createDraft._id})
		})
	}

	render() {
		if (this.state.newDraftID) {
			return <Redirect push={true} to={`/edit/${this.state.newDraftID}`}/>;
		}

		return (
			<div>
				<DraftList/>
				<Button
					raised="raised"
					color="primary"
					onClick={this.handleCreate}
					label='draftCreate'>
					CREATE NEW DRAFT
				</Button>
			</div>
		)
	}

}

TheUserDraft.propTypes = {
	createDraft: PropTypes.func.isRequired
}

export const WithCreateDraft = graphql(createDraftMutation, {
	props: ({mutate}) => ({
		createDraft: () => mutate({
			refetchQueries: [
				{
					query: draftsListQuery
				}
			]
		})
	})
})

export default WithCreateDraft(TheUserDraft)