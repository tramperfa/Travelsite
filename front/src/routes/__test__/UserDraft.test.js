import React from 'react';
import renderer from 'react-test-renderer';
// Note: test renderer must be required after react-native. i
import {shallow, mount} from 'enzyme';
import {MemoryRouter, StaticRouter} from 'react-router-dom';
import toJson from 'enzyme-to-json'

import {UserDraft} from '../UserDraft';
import {MockedProvider} from 'react-apollo/test-utils';

const mockedData = {
	data: {
		createDraft: {
			_id: 'test000',
			title: 'trip to test0',
			lastUpdate: 'Thu Dec 07 2017 04:49:49 GMT+0000 (UTC)'
		}
	}
}
// test('renders correctly', () => { const wrapper = shallow( 		<UserDraft
// createDraft={() => { 				return mockedData 			}}/>, 	);
// expect(wrapper).toMatchSnapshot();   const myButton
// = wrapper.find('withStyles(Button)'); console.log(myButton);
// wrapper.find('button').simulate('click');
//
// });

test('Mount renders correctly', () => {
	const wrapper = mount(
		<MockedProvider>
			<StaticRouter context={{}}>
				<UserDraft createDraft={() => {
						return mockedData
					}}/>
			</StaticRouter>
		</MockedProvider>

	);

	//console.log(JSON.stringify(wrapper));

	wrapper.find('button').forEach(child => {
		console.log("SIMULATED");
		child.simulate('click');
	});
	//expect(handleDelete.calledOnce).toBe(true);
	expect(toJson(wrapper)).toMatchSnapshot();
});
