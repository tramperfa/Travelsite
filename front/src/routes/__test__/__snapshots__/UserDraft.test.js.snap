// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TheUserDraft
    createDraft={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Apollo(draftsList) />,
        <withStyles(Button)
          color="primary"
          label="draftCreate"
          onClick={[Function]}
          raised="raised"
>
          CREATE NEW DRAFT
</withStyles(Button)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "CREATE NEW DRAFT",
          "color": "primary",
          "label": "draftCreate",
          "onClick": [Function],
          "raised": "raised",
        },
        "ref": null,
        "rendered": "CREATE NEW DRAFT",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Apollo(draftsList) />,
          <withStyles(Button)
            color="primary"
            label="draftCreate"
            onClick={[Function]}
            raised="raised"
>
            CREATE NEW DRAFT
</withStyles(Button)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "CREATE NEW DRAFT",
            "color": "primary",
            "label": "draftCreate",
            "onClick": [Function],
            "raised": "raised",
          },
          "ref": null,
          "rendered": "CREATE NEW DRAFT",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
